src/routes/auth/index.ts

```diff
@@ -138,3 +140,41 @@ authRoutes.post('/sign-in', async (context: Context<EnvironmentBindings>) => {
 
 	return context.json(sessionToken);
 });
+
+authRoutes.get('/heartbeat', async (context: Context<EnvironmentBindings>) => {
+	const cookies = context.req.header('Cookie');
+
+	if (!cookies) {
+		return context.json<StatusResponse>({ message: 'No cookies found' }, StatusCodes.UNAUTHORIZED);
+	}
+
+	const sessionToken: string | undefined = getCookie(context, 'auth_token');
+
+	if (!sessionToken) {
+		return context.json<StatusResponse>({ message: 'Invalid or missing token' }, StatusCodes.UNAUTHORIZED);
+	}
+
+	const sessionData = await context.env.SESSION_TOKENS.get<SessionData>(sessionToken, 'json') || undefined;
+
+	if (!sessionData) {
+		return context.json({ message: 'Invalid session' }, StatusCodes.UNAUTHORIZED);
+	}
+
+	if (new Date(sessionData.expiry) < new Date()) {
+		await context.env.SESSION_TOKENS.delete(sessionToken);
+		return context.json({ message: 'Session expired' }, StatusCodes.UNAUTHORIZED);
+	}
+
+	const profile = await getProfileById(context.env.database, sessionData.id);
+
+	if (!profile) {
+		return context.json({ message: 'Internal error getting profile that shoudl exist' }, StatusCodes.INTERNAL_SERVER_ERROR);
+	}
+	const {access} = sessionData;
+	const name = profile.name;
```

Comment:
nitpick: you're de-structuring on line 173 but not on line 174. Personally I perfer de-structuring when possible but I think consistency is more important

Reply:
![image](https://github.com/user-attachments/assets/60a4334a-84c3-436c-85b6-29f5e945b676)